manifests/charts/gateways/istio-egress/templates/autoscale.yaml

```diff
@@ -1,6 +1,7 @@
 {{ $gateway := index .Values "gateways" "istio-egressgateway" }}
 {{- if and $gateway.autoscaleEnabled $gateway.autoscaleMin $gateway.autoscaleMax }}
-apiVersion: autoscaling/v2beta1
+{{- if and .Values.global.autoscalingv2API (semverCompare ">=1.23-0" .Capabilities.KubeVersion.GitVersion)}}
+apiVersion: autoscaling/v2
```

Comment:
I think we still need the v2beta2 option?

* v2beta1: using legacy IOP config. Never used for helm (unless they touch the internal autoscalingv2API flag, but they shouldn't
* v2beta2: anyone else on <1.23
* v2: anyone else >=1.23

Reply:
My original thought was to combine case1 and 2 together, so only if  >=1.23 and not using legacy IOP config would be v2, others are v2beta1. But what you suggest works too, I don't have strong preference